core/src/main/java/com/google/instrumentation/trace/TraceService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+/**
+ * Class that holds the implementation instances for {@link Tracer} and {@link
+ * BinaryPropagationHandler}.
+ *
+ * <p>This implementation of this class is loaded using reflection by the {@link Trace}.
```

Comment:
I am not that familiar with Dagger (or Guice) (too much Java involved). I think it should be fine to allow users to inject the implementation using this frameworks. Do I have to change anything to allow users to do that?